### PR TITLE
Version of benchmark similar to IMB

### PR DESCRIPTION
**Pull request by dkuzmin72**

### CODE REVIEW

**Comment by dkuzmin72** [comment]
I took the heart of benchmark and put it into a cycle to go through all message sizes from 0 to 4M.
I get SegFaults.
I checked with Intel MPI and this code also fails at 4K message size.
Looks like an issue with synchronization.
The benchmark works just fine for one message size.

**Comment by artpol84** [comment]
@dkuzmin72 are you comfortable with the current version? Is it OK to merge?

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #   $HEADER$
 #
 
-CFLAGS ?= -g -lpthread
+CFLAGS ?= -g  -lpthread -DMULTI_MESSAGES
```
Comment (artpol84) [inline Makefile]: @dkuzmin72 I think that multi-message should be the main and only mode for this benchmark.
Single-message was just an intermediate step.
so let's make it a default, so no need for the MULTI_MESSAGES macro
```suggestion
CFLAGS ?= -g  -lpthread
```

File: generic.c [hunk 2]
```diff
@@ -24,6 +24,7 @@ int verify_mode = 0;
 int send_alignment = 1;
 int recv_alignment = 1;
 int buf_unit_size, remote_buf_unit_size;
+int generate_graphs = 0;
```
Comment (artpol84) [inline generic.c:27]: Thanks!

**Review by artpol84 — APPROVED** [review]

**Review by artpol84 — CHANGES_REQUESTED** [review]
Actually, let's get rid of the macro first

File: Makefile (hunk 1 above)
Comment (dkuzmin72) [reply Makefile]: Yep, that's OK!

**Comment by dkuzmin72** [comment]
Did you mean use a command line option instead of MULTI_MESSAGES macro?

**Comment by artpol84** [comment]
@dkuzmin72, correct

**Comment by artpol84** [comment]
Thanks!

**Review by artpol84 — APPROVED** [review]

**Pull request merged**